CodeModel/src/main/java/org/openzen/zenscript/codemodel/type/member/InterfaceResolvedType.java

```diff
@@ -0,0 +1,150 @@
+		return baseType.findSuffixConstructor(suffix);
+	}
+
+	@Override
+	public Optional<InstanceCallableMethod> findCaster(TypeID toType) {
+		if (implementedInterfaces.contains(toType)) {
+			ImplementationMember.InterfaceCaster interfaceCaster = new ImplementationMember.InterfaceCaster(new FunctionHeader(toType), new ImplementationMemberInstance(toType));
+			return Optional.of(interfaceCaster);
+		}
+
+		return findFirstInLocalOrImplementedInterfaces(resolvedType -> resolvedType.findCaster(toType));
+	}
+
+	@Override
+	public Optional<StaticCallable> findStaticMethod(String name) {
+		return baseType.findStaticMethod(name);
+	}
+
+	@Override
+	public Optional<StaticCallable> findStaticGetter(String name) {
+		return baseType.findStaticGetter(name);
+	}
+
+	@Override
+	public Optional<StaticCallable> findStaticSetter(String name) {
+		return baseType.findStaticSetter(name);
+	}
+
+	@Override
+	public Optional<InstanceCallable> findMethod(String name) {
+		return mergeLocalWithImplementedInterfaces(type -> type.findMethod(name), InstanceCallable::union);
+	}
+
+	@Override
+	public Optional<InstanceCallable> findGetter(String name) {
+		return mergeLocalWithImplementedInterfaces(type -> type.findGetter(name), InstanceCallable::union);
+	}
+
+	@Override
+	public Optional<InstanceCallable> findSetter(String name) {
+		return mergeLocalWithImplementedInterfaces(type -> type.findSetter(name), InstanceCallable::union);
+	}
+
+	@Override
+	public Optional<InstanceCallable> findOperator(OperatorType operator) {
+		return findFirstInLocalOrImplementedInterfaces(type -> type.findOperator(operator));
+	}
+
+	@Override
+	public Optional<Field> findField(String name) {
+		return findFirstInLocalOrImplementedInterfaces(type -> type.findField(name));
+	}
+
+	@Override
+	public Optional<TypeSymbol> findInnerType(String name) {
+		return findFirstInLocalOrImplementedInterfaces(type -> type.findInnerType(name));
+	}
+
+	@Override
+	public Optional<CompilableExpression> getContextMember(String name) {
+		return baseType.getContextMember(name);
+	}
+
+	@Override
+	public Optional<SwitchMember> findSwitchMember(String name) {
+		return baseType.findSwitchMember(name);
+	}
+
+	@Override
+	public List<Comparator> comparators() {
+		return Stream.concat(
+						Stream.of(baseType),
+						implementedInterfaces.stream().map(TypeID::resolve)
+				).flatMap(type -> type.comparators().stream())
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public Optional<IteratorInstance> findIterator(int variables) {
+		return findFirstInLocalOrImplementedInterfaces(type -> type.findIterator(variables));
+	}
+
+	@Override
+	public Optional<StaticCallable> findStaticOperator(OperatorType operator) {
+		return baseType.findStaticOperator(operator);
+	}
+
+	private <T> Optional<T> findFirstInLocalOrImplementedInterfaces(Function<ResolvedType, Optional<T>> mapper) {
+		return Stream.concat(
+						Stream.of(baseType),
+						implementedInterfaces.stream().map(TypeID::resolve)
```

Comment:
Since this only calls TypeID::resolve and we don't have a CompileContext, this implementation does not take expansions on interfaces into account (added a test `JavaNativeInterfaceInheritanceTests#expansionFromParentInterfaceAvailableOnChild` to track this).

Needed for CraftTweaker since we make heavy use of expansions.